File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector <int> ans;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            if (nums[i] < pivot) ans.push_back(nums[i]);
        }
        for (int i = 0; i < n; i++) {
            if (nums[i] == pivot) ans.push_back(nums[i]);
        }
        for (int i = 0; i < n; i++) {
            if (nums[i] > pivot) ans.push_back(nums[i]);
        }
        return ans;
    }
};

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.go
func pivotArray(nums []int, p int) []int {
    a1, a2 := []int{}, []int{}
    pc := 0

    for _, num := range nums {
        if num < p {
            a1 = append(a1, num)
        } else if num > p {
            a2 = append(a2, num)
        } else {
            pc++
        }
    }

    for i := 0; i < pc; i++ {
        a1 = append(a1, p)
    }

    return append(a1, a2...)
}